src/runtime/test_case_runner.js

```diff
@@ -197,11 +204,22 @@ export default class TestCaseRunner {
   }
 
   async runStepHooks(step, hookDefinitions) {
-    return Promise.all(
+    const stepHookResults = await Promise.all(
       hookDefinitions.map(async hookDefinition =>
         this.runStepHook(step, hookDefinition)
       )
     )
+    const errors = stepHookResults.filter(result => result.exception)
+    if (errors && errors.length) {
+      return {
+        exception: 'Errors running step hooks',
+        status: Status.FAILED,
+      }
+    } else {
+      return {
+        status: Status.PASSED,
+      }
+    }
   }
 
   async runStep(step) {
```

Comment:
@charlierudolph BeforeStep/AfterStep are working. I do have questions about stacktrace reporting and duration calculation. 
For duration my current thinking is to calculate the duration for the beforeSteps, the step, and the afterSteps and update that value on the `stepResult` prior to returning. 
I'm interested in hearing what you feel would be the best way to handle multiple errors in the hooks .